src/components/Nav/Client/index.tsx

```diff
@@ -0,0 +1,149 @@
+import { useBreakpointValue } from "@/hooks/useBreakpointValue"
+import { useTranslation } from "@/hooks/useTranslation"
+
+const Menu = dynamic(() => import("../Menu"), {
+  ssr: false,
+  loading: () => (
+    <div className="me-8 flex w-full items-center gap-10 px-6 max-md:hidden">
+      {Array.from({ length: 5 }).map((_, i) => (
+        <Skeleton key={i} className="h-6 w-12 py-2" />
+      ))}
+    </div>
+  ),
+})
+
+const MobileMenuLoading = () => (
+  <Skeleton data-label="mobile-menu" className="ms-2 size-6" />
+)
+
+const MobileNavMenu = dynamic(() => import("../Mobile"), {
+  ssr: false,
+  loading: MobileMenuLoading,
+})
+
+const SearchProvider = dynamic(() => import("../../Search"), {
+  ssr: false,
+  loading: () => (
+    <>
+      <div className="flex items-center gap-6 px-2 max-md:hidden xl:px-3">
+        <Skeleton
+          data-label="search-xl"
+          className="hidden h-6 w-[169px] xl:flex"
+        />
+        <Skeleton data-label="search" className="size-6 xl:hidden" />
+      </div>
+      <div className="flex items-center md:hidden">
+        <Skeleton data-label="search" className="mx-2 size-6" />
+        <MobileMenuLoading />
+      </div>
+    </>
+  ),
+})
+
+const LanguagePicker = dynamic(() => import("../../LanguagePicker"), {
+  ssr: false,
+  loading: () => (
+    // LG skeleton width approximates English "[icon] Languages EN" text width
+    <Skeleton className="h-6 max-md:hidden md:mx-2 md:w-[54px] lg:mx-3 lg:w-[8.875rem]" />
+  ),
+})
+
+const ClientSideNav = () => {
+  const { t } = useTranslation("common")
+  const locale = useLocale()
+
+  const { linkSections } = useNavigation()
+  const { toggleColorMode, ThemeIcon, themeIconAriaLabel } = useThemeToggle()
+
+  const languagePickerRef = useRef<HTMLButtonElement>(null)
+
+  // avoid rendering/adding desktop Menu version to DOM on mobile
+  const desktopScreen = useBreakpointValue({ base: false, md: true })
+
+  return (
+    <>
+      {desktopScreen && (
+        <Menu
+          className="animate-fade-in max-md:hidden"
+          sections={linkSections}
+        />
+      )}
+
+      <div className="flex items-center">
+        <SearchProvider>
+          {({ onOpen }) => {
+            return (
+              <>
+                <SearchInputButton
+                  className="animate-fade-in max-xl:hidden"
+                  onClick={onOpen}
+                />
+                <SearchButton
+                  className="animate-fade-in xl:hidden"
+                  onClick={onOpen}
+                />
+
+                {!desktopScreen && (
+                  <MobileNavMenu
+                    toggleColorMode={toggleColorMode}
+                    linkSections={linkSections}
+                    toggleSearch={onOpen}
+                    className="flex animate-fade-in md:hidden"
+                  />
+                )}
+              </>
+            )
+          }}
+        </SearchProvider>
+
+        {desktopScreen && (
+          <Button
```

Comment:
I think we also need to display a skeleton for this button. Currently, the button is no responsive when clicked in pre-load time, affecting INP and FID. We can tackle this is a separate PR.